runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8280,6 +8285,27 @@ done:;
 	}
 
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	void
+	defaultValueWithUnflattenedFlattenables(J9Class *clazz, j9object_t instance)
+	{
+		J9FlattenedClassCacheEntry * entry;
+		UDATA length = clazz->flattenedClassCache->numberOfEntries;
+		for (UDATA index = 0; index < length; index++) {
+			entry = J9_VM_FCC_ENTRY_FROM_CLASS(clazz, index);
+			if (!J9_ARE_ALL_BITS_SET(J9ClassIsFlattened, entry->clazz->classFlags)) {
+				if (entry->offset == UDATA_MAX) {
+					J9Class *definingClass;
+					J9ROMFieldShape *field;
+					J9ROMNameAndSignature * nameAndSig = entry->nameAndSignature;
+					J9UTF8 *name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
+					J9UTF8 *signature = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+					entry->offset = instanceFieldOffset(_currentThread, clazz, J9UTF8_DATA(name), J9UTF8_LENGTH(name), J9UTF8_DATA(signature), J9UTF8_LENGTH(signature), &definingClass, (UDATA *)&field, 0) + J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread);
+				}
+				_objectAccessBarrier.inlineMixedObjectStoreObject(_currentThread, instance, entry->offset, entry->clazz->flattenedClassCache->defaultValue, false);
```

Comment:
add the `J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread)` to the `inlineMixedObjectStoreObject` call not the `entry->offset`